deepspeed/runtime/zero/stage_1_and_2.py

```diff
@@ -1561,14 +1627,23 @@ def zero_grad(self, set_to_none=False):
         """
         # FP32 grad should never exist.
         # For speed, set model fp16 grad to None by default
+        # zero all pointers to grad tensors
         for group in self.bit16_groups:
             for p in group:
                 if set_to_none:
                     p.grad = None  # epilogue and in step
+                    p.grad_accum = None
+                    p.grad_reduc = None
                 else:
                     if p.grad is not None:
                         p.grad.detach_()
                         p.grad.zero_()
+                    if p.grad_accum is not None:
+                        p.grad_accum.detach_()
```

Comment:
I don't think this necessary since these are our buffers which should never be exposed externally (e.g., to autograd).